docs/adr/002-handle-update.md

```diff
@@ -54,32 +57,33 @@ Suppose we want to update the S3 bucket, retaining the tags already assigned:
 
-One possible approach would be to support a generic purge parameter. In addition, a `state: updated` could be added, allowing the user to explicitly confirm their intention to change the state of the resource.
-
 Another pretty big consequence of this decision is that in the current proposal the new states would be applied across all properties of all resources. Given that the general idea behind pravic is to define a bunch of resources and then have those created at once, this does not leave the user with much control over things.
 
-However, let's take a look at the network resource modules that support different states ([see](https://docs.ansible.com/ansible/latest/network/user_guide/network_resource_modules.html)).
+Hence, one possible approach would be to support a generic purge parameter. In addition, a `state: updated` could be added, allowing the user to explicitly confirm their intention to change the state of the resource.
+
+### Solution #2: Comply to network resource modules
+
+Let's take a look at the network resource modules that support different states ([see](https://docs.ansible.com/ansible/latest/network/user_guide/network_resource_modules.html)).
 
 Relating to the previous example, `state=merged` will allow us to accomplish that in one single task. Conversely, if we want to reinitialize resource tags completely, `state=deleted` allows us to do so. `state=replaced`, on the other hand, allows us to replace a resource's existing information with new one.
 
 In addition, `state=gathered`, will allow us to get information on a specific resource.
 
-One proposal is to comply with network resource modules states and implement those for project pravic when it comes to resource update.
+One proposal is to comply with network resource modules states and implement those for project pravic when it comes to resource update. However, one disadvantage of this approach is that the state strategy (e.g., replaced, merged, absent) alters all the options that are changed in the desired compared to the resource state.
+
+To summarize:
+- `state=present` does resource provisioning, while `state=absent` does only resource deprovisioning.
+- When `state=present` and desired state differs from the resource state, no change is applied to the resource. The already existing resource will only be altered when state is `replaced` or `merged`. The resource is provisioned only when `state=present`. `state=merged` or `state=replaced` won't have any impact on a non existing resource.
+- We could use `state=replaced` with an empty list or None value, like we do with the current collections rather than `state=deleted` to reinitialize resource properties.
```

Comment:
I wonder if we could remove `state=present`? The advantages I see by doing this are:

1. It brings us closer in line with the network resource modules. By removing `present` we are only implementing a subset of network resource states, rather than also adding a new state.
2. Having both `present` and `replaced`/`merged` means that a user would essentially have to know the state of their infrastructure in order to know which one to use. Using `replaced` or `merged` on a resource that does not already exist should result in the same end state, and function similarly to `present`.

Do we have any examples in our current modules where something like an updated state is a no-op when the resource doesn't exist? I can't think of any.